documentation/incremental_settings.md

```diff
@@ -0,0 +1,85 @@
+# Incremental Settings for Ansible Values
+
+In some cases in Ansible, you provide most or all of the pertient attributes
+of the managed object.  For example, for the `file` module, you provide the
+content, permissions, ownership, and other attributes of the file to be
+managed. In other cases, you have to specify how to change the *incremental*
+state of the system.  For example, when you use the `sysctl` module, you do
+not provide every single `sysctl` setting describing the entire state of the
+kernel settings.  Instead, you provide a single setting, which *incrementally*
+updates the state of all of the `sysctl` settings, by adding, modifying, and
+removing single values.
+```yaml
+- name: modify a single kernel setting
+  sysctl:
+    name: some_setting
+    value: 123  # to add/replace the value
+```
+or
+```yaml
+- name: modify a single kernel setting
+  sysctl:
+    name: some_setting
+    state: absent  # to remove the value
+```
```

Comment:
I would change the first phrase to clarify the rationale why we are not following the usual pattern of Ansible roles and playbooks and rather follow the approach commonly taken by modules:
"Many roles in the System Roles project need to follow the usual pattern of module interfaces and allow specifying incremental changes instead of the complete state to be able to be safely used multiple times in a playbook without the later invocation clobbering the result of the previous invocation, and to preserve previous state of the managed systems. We had a need to generalize the usual approach of modules to lists of settings ..."
Alternatively, we can leave it as it is and add the rationale in a future commit.

Reply:
fixed in https://github.com/linux-system-roles/linux-system-roles.github.io/pull/21/commits/373b5ac990e225b3ac11431fef1154a65e047475 - I mostly used your text